pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2803,18 +2819,21 @@ public void readStartupMessages() throws IOException, SQLException {
   public void receiveParameterStatus() throws IOException, SQLException {
     // ParameterStatus
     pgStream.receiveInteger4(); // MESSAGE SIZE
-    String name = pgStream.receiveString();
-    String value = pgStream.receiveString();
+    final String name = pgStream.receiveCanonicalString();
+    final String value = pgStream.receiveCanonicalString();
```

Comment:
@sehrope had a [question](https://github.com/pgjdbc/pgjdbc/pull/1988#discussion_r698596283) in the original PR about whether to canonicalize the values that still needs more feedback.